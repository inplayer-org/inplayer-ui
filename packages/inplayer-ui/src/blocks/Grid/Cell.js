import styled from 'styled-components';

const Cell = styled.div`
  height: 100%;
  min-width: 0;
  grid-column-end: ${({ width = 1 }) => `span ${width}`};
  grid-row-end: ${({ height = 1 }) => `span ${height}`};
  ${({ left }) => left && `grid-column-start: ${left}`};
  ${({ top }) => top && `grid-row-start: ${top}`};
  ${({ center }) => center && `text-align: center`};
  ${({ area }) => area && `grid-area: ${area}`};
  ${({ middle }) =>
    middle &&
    `
    display: inline-flex;
    flex-flow: column wrap;
    justify-content: center;
    justify-self: stretch;
  `};
`;

export default Cell;
