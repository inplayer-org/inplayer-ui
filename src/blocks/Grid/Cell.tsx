import styled from 'styled-components';

interface Props {
  width: number;
  height: number;
  left: string;
  top: string;
  center: string;
  area: string;
  middle: string;
}

const Cell = styled.div<Props>`
  height: 100%;
  min-width: 0;
  grid-column-end: ${({ width = 1 }) => `span ${width}`};
  grid-row-end: ${({ height = 1 }) => `span ${height}`};
  ${({ left }) => left && `grid-column-start: ${left}`};
  ${({ top }) => top && `grid-row-start: ${top}`};
  ${({ center }) => center && `text-align: center`};
  ${({ area }) => area && `grid-area: ${area}`};
  ${({ middle }) =>
    middle &&
    `
    display: inline-flex;
    flex-flow: column wrap;
    justify-content: center;
    justify-self: stretch;
  `};
`;

export default Cell;
