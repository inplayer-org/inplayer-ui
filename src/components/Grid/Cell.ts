import styled, { CSSProperties } from 'styled-components';

type CellProps = {
  width?: CSSProperties;
  height?: CSSProperties;
  left?: CSSProperties;
  top?: CSSProperties;
  center?: CSSProperties;
  area?: CSSProperties;
  middle?: CSSProperties;
};

const Cell = styled.div<CellProps>`
  height: 100%;
  min-width: 0;
  grid-column-end: ${({ width = 1 }) => `span ${width}`};
  grid-row-end: ${({ height = 1 }) => `span ${height}`};
  ${({ left }) => left && `grid-column-start: ${left}`};
  ${({ top }) => top && `grid-row-start: ${top}`};
  ${({ center }) => center && `text-align: center`};
  ${({ area }) => area && `grid-area: ${area}`};
  ${({ middle }) =>
    middle &&
    `
    display: inline-flex;
    flex-flow: column wrap;
    justify-content: center;
    justify-self: stretch;
  `};
`;

export default Cell;
