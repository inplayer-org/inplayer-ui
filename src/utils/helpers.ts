import moment from 'moment';

export const getMonthOptions = (): Array<any> =>
  moment.months().map((label, value) => ({ value, displayName: label }));

export const getYearOptions = () => {
  const yearOptions: Array<any> = [];
  for (let i = moment().year() - 100; i <= moment().year(); i += 1) {
    yearOptions.push({ value: i.toString(), displayName: i.toString() });
  }

  return yearOptions;
};
