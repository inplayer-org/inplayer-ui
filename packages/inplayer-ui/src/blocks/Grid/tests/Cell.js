import React from 'packages/inplayer-ui/src/blocks/Grid/tests/react';
import { renderWithTheme } from 'packages/inplayer-ui/src/blocks/Grid/tests/helpers';
import Cell from '../Cell';
import 'packages/inplayer-ui/src/blocks/Grid/tests/jest-styled-components';

describe('Grid.Column Element', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(<Cell />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
