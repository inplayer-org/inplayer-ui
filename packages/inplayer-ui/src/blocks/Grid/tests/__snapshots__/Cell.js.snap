// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid.Column Element renders correctly 1`] = `
.c0 {
  height: 100%;
  min-width: 0;
  grid-column-end: span 1;
  grid-row-end: span 1;
}

<div
  className="c0"
/>
`;
